Ensure rsync is installed on the guest if needed

psss opened the pull request:
To sync workdir `rsync` is needed on both host & guest.
Move the `rsync` requires to the main `tmt` package.
Implement a `requires()` method for the provision step.
Update `local` and `podman` plugins accordingly.
Allow to `login` even if `rsync` is not installed.
Adjust `Guest.execute()` to accept `cwd=None` as well.

Resolves #437 and fixes #558 as well.

lukaszachy reviewed (commented):
Isn't better place for 'rsync' require in Guest class? We need it for its push/pull methods. 
So 'provision' plugin should ask its Guest class for requires instead of having them hard coded.

psss commented:
> Isn't better place for 'rsync' require in Guest class?

Very good point, thanks @lukaszachy. I've updated the code accordingly. I've also fixed the `login` use case so that when `rsync` is not available on the guest it only gives a warning and uses the default `cwd` for the user. Please, review.

lukaszachy reviewed (approved):
Very nice. `tmt run provision --how container prepare login` doesn't need rsync so it is not installed nor warning is printed.
`tmt run provision --how connect --guest FQDN prepare login` does need rsync so it is installed but only if 'prepare' is there.

psss commented:
Great, thanks for the review!
